### DISCUSSION TITLE
How to render an array of object?

### DISCUSSION THREAD

**Discussion by aztack** [pmndrs/leva#404] [Q&A]
When I try to render a object with a array property, it complains: `input at path `list` is not recognized`

```jsx
import "./styles.css";
import { useControls } from "leva";

function MyComponent(props) {
  useControls({ id: 123, list: props.list }); // <---- list is not recognized
  return <></>;
}

export default function App() {
  const group1 = [
    { name: "a", age: 12 },
    { name: "b", age: 24 }
  ];
  return (
    <div className="App">
      <MyComponent list={group1} />
    </div>
  );
}

```
[codesandbox](https://codesandbox.io/s/leva-render-array-rfssz6?file=/src/App.js)

So what is the proper way to render a list of objects?

**Comment by dbismut** [comment]
This is a question. Not an issue. How do you expect Leva to render an array of objects?

**Reply by aztack** [reply]
Maybe something like this:
<img width="205" alt="image" src="https://user-images.githubusercontent.com/782871/199282441-f5131830-1ead-4f50-8fa9-d268fd789134.png">

What I want is that array of objects rendered in a folder and items in that array can be deleted/added dynamically.
Is this possible?

**Reply by dbismut** [reply]
Not at the moment but that could be doable through a plugin maybe.

**Reply by aztack** [reply]
<del>Since `useControls` inside a `forEach` or `for loop` will result in
`ERROR Warning: React has detected a change in the order of Hooks called by ...`
when array size changed.
Is there a JSX way to use leva controls to avoid this error?</del>

Never mind, I found JSX example in demo/leva-custom-plugin folder.
